src/main/resources/templates/mdBenefitsFlow/mailingAddress.html

```diff
@@ -24,8 +24,7 @@
                 <th:block th:replace="~{fragments/inputs/address ::
                                   address(
                                     validate=true,
-                                    inputName='mailingAddress',
-                                    streetAddress2HelpText=#{general.optional}
+                                    inputName='mailingAddress'
```

Comment:
Is there a compelling technical reason to have two separate pages for the scenarios?  From what I can tell, the only difference between this and the `mailingAddressNoAddresse` page is some JS and the logic to add the checkbox to use homeAddress. 

I realize this was in before you worked on this ticket, but do you know of a good reason to not just have these two be one page where the checkbox is conditionally shown?

Reply:
I was thinking the very same thing and decided to stick with the existing pattern so as to minimize change.

Reply:
Okay, that works for me.  For now we can leave it.

Reply:
@bseeger and I decided to clean this up.

Reply:
Thanks so much for doing this, @vrajmohan! I think it will prevent confusion in the future